extend.php

```diff
@@ -58,5 +59,6 @@
         }),
 
     (new Flarum\Settings())
-        ->default('fof-sentry.monitor_performance', 0),
+        ->default('fof-sentry.monitor_performance', 0)
+        ->default('fof-sentry.environment', resolve(UrlGenerator::class)->to('forum')->base()),
```

Comment:
I believe we had issues in the past when resolving `UrlGenerator` early in Listener `__construct` methods. 
I can't find all the info regarding this issue, but I did find this commit from Franz in flarum/mentions (https://github.com/flarum/mentions/commit/4dc8068fc335d99b630ac0915cd2ffa67c696f4e) that stopped resolving UrlGenerator early in an event subscriber class.

Reply:
I've had a chat with @imorland and he mentioned that for this to work, the `flarum\core` has to be `^1.3.1`. Could this be linked to the issue mentioned here? https://github.com/flarum/framework/pull/3439

Reply:
No, the 1.3.1 issue is different to this. @datitisev is referring to the early resolution of the `UrlGenerator` here.

The < 1.3.1 issue is chaining multiple `->default` settings were not registered correctly

Reply:
No. The UrlGenerator resolution problem here isn't even related to the Settings extender itself, as you are calling `resolve` as a parameter to `->default()`. It doesn't matter how the extender handles that output as the resolution occurs in `extend.php` before the extender does anything.

I'm not entirely sure when the Sentry `init` function is called in the lifecycle, so I don't know if that would be after app boot. It might work there or it might need code in boot to run the Sentry `init` (I don't know if this can be run multiple times).

Reply:
Would you prefer if I move the UrlGenerator default to the `SentryJavaScript.php` and `SentryServiceProvider.php` files if no custom environment is set in the admin settings?

Reply:
Yep. Looks like that should work.

I added some logs to see the run order and SentryJavascript runs well after boot. And the Sentry `init` code only runs when Sentry is actually triggered (e.g. by an error, I had a `SentryServiceProvider@register HubInterface` log line that never appeared) so it should be fine.

```
[2022-07-28 13:29:43] flarum.DEBUG: SentryServiceProvider@register
[2022-07-28 13:29:43] flarum.DEBUG: SentryServiceProvider@boot
[2022-07-28 13:29:44] flarum.DEBUG: SentryJavascript
```

Reply:
Does this work for you?

Reply:
@datitisev just tell me when I can do something her

Reply:
@PatrickRoethlisberger I think this should work.

@imorland could you check as well whether the fix should solve the issues? 
I mainly checked load order but didn't test it with an ext that would be broken with the previous code.